scripts/lib/upgrade-zulip-stage-2

```diff
@@ -295,109 +355,137 @@ else:
         preexec_fn=su_to_zulip,
     )
 
+# Perform system checks -- including database checks, so we don't need
+# to do them when we do migrations, below.
+subprocess.check_call(["./manage.py", "check", "--database", "default"], preexec_fn=su_to_zulip)
+
 # Our next optimization is to check whether any migrations are needed
 # before we start the critical section of the restart.  This saves
-# about 1s of downtime in a no-op upgrade.
+# about 1s of downtime in a no-op upgrade.  We omit this check if we
+# already stopped the server above, due to low memory.
 migrations_needed = False
-if not args.skip_migrations:
+if not IS_SERVER_UP:
+    migrations_needed = True
+elif not args.skip_migrations:
     logging.info("Checking for needed migrations")
     migrations_output = subprocess.check_output(
-        ["./manage.py", "showmigrations"], preexec_fn=su_to_zulip, text=True
+        ["./manage.py", "showmigrations", "--skip-checks"], preexec_fn=su_to_zulip, text=True
     )
     for ln in migrations_output.split("\n"):
         line_str = ln.strip()
         if line_str.startswith("[ ]"):
             migrations_needed = True
 
-# NOTE: Here begins the most likely critical period, where we may be
-# shutting down the server; we should strive to minimize the number of
-# steps that happen between here and the "Restarting Zulip" line
-# below.
-
-if rabbitmq_dist_listen:
-    shutdown_server()
-    logging.info("Shutting down rabbitmq to adjust its ports...")
-    subprocess.check_call(["/usr/sbin/service", "rabbitmq-server", "stop"])
-
-if cookie_size is not None and cookie_size == 20:
-    # Checking for a 20-character cookie is used as a signal that it
-    # was generated by Erlang's insecure randomizer, which only
-    # provides between 20 and 36 bits of entropy; were it 20
-    # characters long by a good randomizer, it would be 96 bits and
-    # more than sufficient.  We generate, using good randomness, a
-    # 255-character cookie, the max allowed length.
-    shutdown_server()
-    logging.info("Generating a secure erlang cookie...")
-    subprocess.check_call(["./scripts/setup/generate-rabbitmq-cookie"])
+if minimal_change and migrations_needed:
+    logging.error("Would need to apply migrations --aborting!")
```

Comment:
I think this is meant to be `migrations -- aborting` like the other similar errors.